diff --git a/node_modules/y-webrtc/src/y-webrtc.js b/node_modules/y-webrtc/src/y-webrtc.js
index 609328a..87b425c 100644
--- a/node_modules/y-webrtc/src/y-webrtc.js
+++ b/node_modules/y-webrtc/src/y-webrtc.js
@@ -30,7 +30,7 @@ const messageBcPeerId = 4
 /**
  * @type {Map<string, SignalingConn>}
  */
-const signalingConns = new Map()
+export const signalingConns = new Map()
 
 /**
  * @type {Map<string,Room>}
@@ -473,89 +473,98 @@ const publishSignalingMessage = (conn, room, data) => {
   }
 }
 
-export class SignalingConn extends ws.WebsocketClient {
-  constructor (url) {
-    super(url)
-    /**
-     * @type {Set<WebrtcProvider>}
-     */
-    this.providers = new Set()
-    this.on('connect', () => {
-      log(`connected (${url})`)
-      const topics = Array.from(rooms.keys())
-      this.send({ type: 'subscribe', topics })
-      rooms.forEach(room =>
-        publishSignalingMessage(this, room, { type: 'announce', from: room.peerId })
-      )
-    })
-    this.on('message', m => {
-      switch (m.type) {
-        case 'publish': {
-          const roomName = m.topic
-          const room = rooms.get(roomName)
-          if (room == null || typeof roomName !== 'string') {
+/**
+ * @param {SignalingConn} conn
+ */
+export const setupSignalingHandlers = (conn) => {
+  conn.on('connect', () => {
+    log(`connected (${conn.url})`)
+    const topics = Array.from(rooms.keys())
+    conn.send({ type: 'subscribe', topics })
+    rooms.forEach(room =>
+      publishSignalingMessage(conn, room, { type: 'announce', from: room.peerId })
+    )
+  })
+  conn.on('message', m => {
+    switch (m.type) {
+      case 'publish': {
+        const roomName = m.topic
+        const room = rooms.get(roomName)
+        if (room == null || typeof roomName !== 'string') {
+          return
+        }
+        const execMessage = data => {
+          const webrtcConns = room.webrtcConns
+          const peerId = room.peerId
+          if (data == null || data.from === peerId || (data.to !== undefined && data.to !== peerId) || room.bcConns.has(data.from)) {
+            // ignore messages that are not addressed to this conn, or from clients that are connected via broadcastchannel
             return
           }
-          const execMessage = data => {
-            const webrtcConns = room.webrtcConns
-            const peerId = room.peerId
-            if (data == null || data.from === peerId || (data.to !== undefined && data.to !== peerId) || room.bcConns.has(data.from)) {
-              // ignore messages that are not addressed to this conn, or from clients that are connected via broadcastchannel
-              return
-            }
-            const emitPeerChange = webrtcConns.has(data.from)
-              ? () => {}
-              : () =>
-                room.provider.emit('peers', [{
-                  removed: [],
-                  added: [data.from],
-                  webrtcPeers: Array.from(room.webrtcConns.keys()),
-                  bcPeers: Array.from(room.bcConns)
-                }])
-            switch (data.type) {
-              case 'announce':
-                if (webrtcConns.size < room.provider.maxConns) {
-                  map.setIfUndefined(webrtcConns, data.from, () => new WebrtcConn(this, true, data.from, room))
-                  emitPeerChange()
-                }
-                break
-              case 'signal':
-                if (data.signal.type === 'offer') {
-                  const existingConn = webrtcConns.get(data.from)
-                  if (existingConn) {
-                    const remoteToken = data.token
-                    const localToken = existingConn.glareToken
-                    if (localToken && localToken > remoteToken) {
-                      log('offer rejected: ', data.from)
-                      return
-                    }
-                    // if we don't reject the offer, we will be accepting it and answering it
-                    existingConn.glareToken = undefined
+          const emitPeerChange = webrtcConns.has(data.from)
+            ? () => {}
+            : () =>
+              room.provider.emit('peers', [{
+                removed: [],
+                added: [data.from],
+                webrtcPeers: Array.from(room.webrtcConns.keys()),
+                bcPeers: Array.from(room.bcConns)
+              }])
+          switch (data.type) {
+            case 'announce':
+              if (webrtcConns.size < room.provider.maxConns) {
+                map.setIfUndefined(webrtcConns, data.from, () => new WebrtcConn(conn, true, data.from, room))
+                emitPeerChange()
+              }
+              break
+            case 'signal':
+              if (data.signal.type === 'offer') {
+                const existingConn = webrtcConns.get(data.from)
+                if (existingConn) {
+                  const remoteToken = data.token
+                  const localToken = existingConn.glareToken
+                  if (localToken && localToken > remoteToken) {
+                    log('offer rejected: ', data.from)
+                    return
                   }
-                }
-                if (data.signal.type === 'answer') {
-                  log('offer answered by: ', data.from)
-                  const existingConn = webrtcConns.get(data.from)
+                  // if we don't reject the offer, we will be accepting it and answering it
                   existingConn.glareToken = undefined
                 }
-                if (data.to === peerId) {
-                  map.setIfUndefined(webrtcConns, data.from, () => new WebrtcConn(this, false, data.from, room)).peer.signal(data.signal)
-                  emitPeerChange()
+              }
+              if (data.signal.type === 'answer') {
+                log('offer answered by: ', data.from)
+                const existingConn = webrtcConns.get(data.from)
+                if (existingConn) {
+                  existingConn.glareToken = undefined
                 }
-                break
-            }
+              }
+              if (data.to === peerId) {
+                map.setIfUndefined(webrtcConns, data.from, () => new WebrtcConn(conn, false, data.from, room)).peer.signal(data.signal)
+                emitPeerChange()
+              }
+              break
           }
-          if (room.key) {
-            if (typeof m.data === 'string') {
-              cryptoutils.decryptJson(buffer.fromBase64(m.data), room.key).then(execMessage)
-            }
-          } else {
-            execMessage(m.data)
+        }
+        if (room.key) {
+          if (typeof m.data === 'string') {
+            cryptoutils.decryptJson(buffer.fromBase64(m.data), room.key).then(execMessage)
           }
+        } else {
+          execMessage(m.data)
         }
       }
-    })
-    this.on('disconnect', () => log(`disconnect (${url})`))
+    }
+  })
+  conn.on('disconnect', () => log(`disconnect (${conn.url})`))
+}
+
+export class SignalingConn extends ws.WebsocketClient {
+  constructor (url) {
+    super(url)
+    /**
+     * @type {Set<WebrtcProvider>}
+     */
+    this.providers = new Set()
+    setupSignalingHandlers(this)
   }
 }
 
